### ISSUE TITLE
reorder messages on the fly

### ISSUE THREAD

**Issue by maddrag0n** [be-gosu/issues#107]
when you send a message and someone else writes another message within the delay that it takes for the server to register your message, message ordering is wrong. your client renders your message optimistically and therefore shows your message first, even tho your message reached the server after the other message.

client should update message ordering for own messages when it receives the echo from the server.

**Comment by DanielHeckrath** [comment]
I don't think this is the right approach. Message reordering can never be consistent between reloads, and is therefore not the right way to do it. 

We still haven't rolled out our feature to mark optimistic messages as pending, or sending. I feel like marking the messages that way, would clarify why messages might be reordered.

**Comment by maddrag0n** [comment]
messages are never reordered on the fly tho!
